--- origsrc/gossip-0.29/configure.ac	2008-05-08 04:02:48.000000000 -0500
+++ src/gossip-0.29/configure.ac	2008-05-15 06:35:39.750000000 -0500
@@ -12,6 +12,10 @@
 AC_PROG_CC
 AC_HEADER_STDC
 
+AC_CHECK_FUNCS(getaddrinfo, [], 
+	[AC_CHECK_LIB(getaddrinfo, getaddrinfo, [], 
+		[AC_MSG_ERROR([getaddrinfo is required to build $PACKAGE])])])
+
 AM_PROG_LIBTOOL
 AM_PATH_GLIB_2_0 
 AC_PATH_XTRA
--- origsrc/gossip-0.29/libloudermouth/lm-bs-client.c	2008-05-08 02:48:17.000000000 -0500
+++ src/gossip-0.29/libloudermouth/lm-bs-client.c	2008-05-15 06:35:39.765625000 -0500
@@ -46,6 +46,10 @@
 #include "lm-bs-session.h"
 #include "lm-bs-client.h"
 
+#ifndef HAVE_GETADDRINFO
+#include <getaddrinfo.h>
+#endif
+
 #define IO_READ_FLAGS    G_IO_IN | G_IO_PRI | G_IO_HUP | G_IO_ERR
 #define IO_WRITE_FLAGS   G_IO_OUT | G_IO_ERR
 #define IO_CONNECT_FLAGS G_IO_OUT | G_IO_ERR
